src/main/java/ch/njol/skript/bukkitutil/ComponentWrapper.java

```diff
@@ -0,0 +1,106 @@
+package ch.njol.skript.bukkitutil;
+
+import ch.njol.skript.util.ItemSource;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.converter.Converter;
+
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * Used for storing the component and/or the item.
+ * @param <T> The type of component
+ */
+public abstract class ComponentWrapper<T> {
+
+	private @Nullable ItemSource itemSource;
+	private @Nullable ItemStack itemStack;
+	private final T component;
+
+	/**
+	 * Create a {@link ComponentWrapper} that converts {@code itemStack} into an {@link ItemSource}.
```

Comment:
Docs need improvement here. The component wrapper isn't converting itemstacks to itemsources. it's a wrapper for a component. Describe the result the user gets from the constructor, not the internals of the code. What do I get in return from an itemstack? Why is it different from the other constructors?